tests/foreman/destructive/test_rename.py

```diff
@@ -112,9 +123,23 @@ def test_positive_rename_satellite(module_org, module_product, module_target_sat
         'repository published path not updated correctly'
     )
 
-    # check for any other occurences of old hostname
-    result = module_target_sat.execute(f'grep " {old_hostname}" --exclude-dir="promtail" /etc/* -r')
-    assert result.status != 0, 'there are remaining instances of the old hostname'
+    # check config files (except certs/keys and /etc/template) for occurences of old hostname
+    result = module_target_sat.execute(
+        f'grep "{old_hostname}" /etc -r --exclude=template --exclude-dir={{promtail,pki}} --exclude=*.{{pem,cert,bak}}'
+    )
+    assert result.status != 0, (
+        'there are remaining instances of the old hostname in the config files'
+    )
+    # check certs for occurences of new hostname in Subject
+    assert all(
```

Comment:
The point I tried to make in https://github.com/SatelliteQE/robottelo/pull/18142#discussion_r2035053640 was that you want to see the data when it fails but with this I'm afraid it's swallowed internally. That's why I explicitly created a list of parsed certificates and then filtered it down. I think the following below would probably give you that and which file failed the test because pytest shows the local variables.
```python
certs = [(path, module_target_sat.execute(f'openssl x509 -text -in {cert}').stdout) for cert in new_certs]
regex = re.compile(f"Subject:.*{re.escape(new_hostname)}")
assert not [(path, cert) for (path, cert) in certs if not regex.match(cert)], 'missing new hostname on certificate'
```

Reply:
What about this? 
```
E       AssertionError: there is missing new hostname in some of the certs
E       assert not ['/etc/foreman/client_cert.pem', '/etc/foreman-proxy/ssl_cert.pem']
```
Just lets not output certificate itself

Reply:
Changed.
Assertion is now more introspective (^ see above)